### DISCUSSION TITLE
How can I copy each calendar entry to my local calendar?

### DISCUSSION THREAD

**Discussion by gubrist** [bitfireAT/icsx5#125] [Q&A]
I export my Strava calendar with Strautomator

https://strautomator.com/feature/calendar

However, there is a time limit of 180 days in the past. 

ICSx⁵
helps me to import this calendar. :-) 

How can I copy each calendar entry to my local calendar?

**Comment by ArnyminerZ** [comment]
I'm not sure what do you mean by "copy each calendar entry to my local calendar", can you reformulate?

**Comment by gubrist** [comment]
Yes, of course:

![Screenshot_20230227_090942_Calendar](https://user-images.githubusercontent.com/22197180/221509283-df4c3fcb-b83d-4c18-bb5e-2acbe8cdc8d9.jpg)

I only use my local calendar, which is synched locally via WiFi / Bluetooth with my PC with My Phone Explorer MPE  (https://www.fjsoft.at/en/). 

The (timely limited) subscribed calendar entries with my sports activities are read only so I cannot copy them to my local calendar, so far.

**Reply by ArnyminerZ** [reply]
So, if I understand correctly, you want to use ICSx⁵ to copy the subscriptions from Strautomator to MPE?

**Comment by gubrist** [comment]
No, I want to copy them to my local phone calendar.

**Reply by ArnyminerZ** [reply]
Then you just have to follow the setup instructions at https://icsx5.bitfire.at/usage/

**Reply by gubrist** [reply]
That is the way I have set it up but leads into a separate subscribed - and not a local - calendar.

The calendar entries are read only and cannot be copied or moved into the local calendar.

**Reply by ArnyminerZ** [reply]
ICSx⁵ is used just for synchronization, all the calendars are read-only. It's for subscribing to ics streams. For allowing modification you should use DAVx⁵

**Reply by gubrist** [reply]
> For allowing modification you should use DAVx⁵

Thanks.

Will the modification of the subscribed and downloaded entries be posssible?

( https://strautomator.com/feature/calendar will not modify the strava calendar.)

Will DAVx⁵ free work as well as I have purchased the ICSx⁵ in vain.

**Reply by ArnyminerZ** [reply]
> Will the modification of the subscribed and downloaded entries be posssible?

Not in ICSx⁵. The purpose of the app is to subscribe to streams, such as the university's schedule, or as in your example, your Strava schedule.

However, I don't think that strautomator supports editing, it says that it exports `ics`, which is exactly the purpose of ICSx⁵, to subscribe to these kind of urls. But, by definition, these URLs are read-only, for editing you should have a DAV server. I'm not sure if you are aware of these terms.

**Reply by gubrist** [reply]
I understand the basic concept of DAV. 

However, will the:

- modification of the subscribed and downloaded entries

or

- copy  of the subscribed and downloaded entries
 
with DAVx⁵ free  be posssible?

**Reply by ArnyminerZ** [reply]
> modification of the subscribed and downloaded entries

Yes

> copy of the subscribed and downloaded entries

If by "copy" you mean synchronizing between your server and your Android device, yes.

In both cases, of course considering that your server supports DAV.

**Comment by gubrist** [comment]
I have installed 

DAVx⁵ 

but it does not accept - as you had mentioned - 
the provided link ending.ics

https://strautomator.com/api/calendar/1562xxxxxxxxxx/all.ics?

I may conclude my investigation with the statement that strautomator.com does not support DAV.

**Reply by ArnyminerZ** [reply]
I guess it's just for getting your events, not for creating them. For that you should use the Strava app.

**Reply by gubrist** [reply]
The Strava App does not provide / export calendar entries. 

That is why strautomator.com provides them as iCal.

**Reply by ArnyminerZ** [reply]
But Strava is who creates the events? I have never used Strava very much 😅
